client/data/plugins/use-monitor-settings-mutation.ts

```diff
@@ -46,20 +47,49 @@ export const useMonitorSettingsQuery = (
 	} );
 };
 
+interface MonitorSettingVariables {
+	retryFlag?: boolean;
+}
+
 export function useCreateMonitorSettingsMutation( siteSlug: SiteSlug, queryOptions = {} ) {
+	const MAX_RETRIES = 3;
+	let retryCount = 0;
+
+	const isMonitorNotActive = ( data: UpdateMonitorSettingsCreate ) => {
+		return data?.settings && data?.settings?.monitor_active === false;
+	};
+
 	const mutation = useMutation( {
-		mutationFn: ( params: object ) =>
+		mutationFn: ( params: MonitorSettingVariables ) =>
 			wpcomRequest( {
 				path: `/sites/${ siteSlug }/jetpack-monitor-settings`,
 				apiNamespace: 'wpcom/v2',
 				method: 'POST',
 				body: params,
-			} ),
+			} ) as Promise< UpdateMonitorSettingsCreate >,
 		...queryOptions,
+		onSuccess: ( data, variables ) => {
+			const { retryFlag } = variables;
+			if ( retryFlag && isMonitorNotActive( data ) ) {
+				if ( retryCount < MAX_RETRIES ) {
```

Comment:
Can we adapt this function to rely on the default retry functionality following [Tanstack documentation](https://tanstack.com/query/latest/docs/framework/react/reference/useMutation)?
The logic looks like an overhead.

Reply:
Thanks for the feedback. The reason I did this on the onSuccess callback is because `retry` option doesn't handle success responses so it won't be triggered. Eventually, I made the API call async here so we could capture the response and throw an error earlier. It relies on the `retry` and `retryDelay` now.